TriblerGUI/tribler_window.py

```diff
@@ -301,7 +302,7 @@ def on_low_storage(self):
         close_dialog.show()
 
     def on_torrent_finished(self, torrent_info):
-        if self.tray_icon:
+        if self.tray_icon and not sip.isdeleted(self.tray_icon):
```

Comment:
This won't work. We've had this before. As the deletion of objects in Qt is threaded, in between this check and the showMessage, the tray icon can be removed. Instead, you should catch a `RuntimeError` every time you interface with the tray icon. We do this in other places as well.

Reply:
And.. it's merged.

Reply:
Well, I thought it solved the problem. I will revert it and fix it properly then I guess.

Reply:
See this:

https://github.com/Tribler/tribler/blob/4cec53cced28608efb5884ab9d02556c7234b6b7/TriblerGUI/widgets/downloadspage.py#L86

Reply:
I just did a search, the market community is apparently also interfacing with the `tray_icon` directly. This should also be changed.